src/cls/rgw/cls_rgw_types.cc

```diff
@@ -577,7 +577,7 @@ void rgw_bucket_dir::dump(Formatter *f) const
   f->open_object_section("header");
   header.dump(f);
   f->close_section();
-  map<string, rgw_bucket_dir_entry>::const_iterator iter = m.begin();
+  auto iter = m.cbegin();
```

Comment:
since we are modifying anyway we could change the for loop below and the others to a range based one?

Reply:
It looks like there could be opportunities for const versions in the loops before -- either cbegin/cend or, following @theanalyst 's lead, const range loops).

Reply:
good catch on cbegin(), cend();  I would prefer range-for all over the place, but I think that's a different cleanup.

Matt